src/connectors/index.ts

```diff
@@ -1,49 +1,126 @@
+import Fortmatic from 'fortmatic'
 
 import UNISWAP_LOGO_URL from '../assets/svg/logo.svg'
-import getLibrary from '../utils/getLibrary'
-import { FortmaticConnector } from './Fortmatic'
-import { NetworkConnector } from './NetworkConnector'
 
-const FORMATIC_KEY = process.env.REACT_APP_FORTMATIC_KEY
+export enum Wallet {
+  INJECTED = 'INJECTED',
+  COINBASE_WALLET = 'COINBASE_WALLET',
+  WALLET_CONNECT = 'WALLET_CONNECT',
+  FORTMATIC = 'FORTMATIC',
+}
+
+export const WALLETS = [Wallet.COINBASE_WALLET, Wallet.WALLET_CONNECT, Wallet.INJECTED, Wallet.FORTMATIC]
+
+export const getWalletForConnector = (connector: Connector) => {
+  switch (connector) {
+    case injected:
+      return Wallet.INJECTED
+    case coinbaseWallet:
+      return Wallet.COINBASE_WALLET
+    case walletConnect:
+      return Wallet.WALLET_CONNECT
+    case fortmatic:
+      return Wallet.FORTMATIC
+    default:
+      throw Error('unsupported connector')
+  }
+}
+
+export const getConnectorForWallet = (wallet: Wallet) => {
+  switch (wallet) {
+    case Wallet.INJECTED:
+      return injected
+    case Wallet.COINBASE_WALLET:
+      return coinbaseWallet
+    case Wallet.WALLET_CONNECT:
+      return walletConnect
+    case Wallet.FORTMATIC:
+      return fortmatic
+  }
+}
+
+const getConnectorListItemForWallet = (wallet: Wallet) => {
+  return {
+    connector: getConnectorForWallet(wallet),
+    hooks: getHooksForWallet(wallet),
+  }
+}
 
-export const network = new NetworkConnector({
-  urls: INFURA_NETWORK_URLS,
-  defaultChainId: 1,
-})
+export const getHooksForWallet = (wallet: Wallet) => {
+  switch (wallet) {
+    case Wallet.INJECTED:
+      return injectedHooks
+    case Wallet.COINBASE_WALLET:
+      return coinbaseWalletHooks
+    case Wallet.WALLET_CONNECT:
+      return walletConnectHooks
+    case Wallet.FORTMATIC:
+      return fortmaticHooks
+  }
+}
+
+export const [network, networkHooks] = initializeConnector<Network>(
+  (actions) => new Network(actions, INFURA_NETWORK_URLS, true, 1),
+  Object.keys(INFURA_NETWORK_URLS).map((chainId) => Number(chainId))
+)
+
+export const [injected, injectedHooks] = initializeConnector<MetaMask>(
+  (actions) => new MetaMask(actions),
+  ALL_SUPPORTED_CHAIN_IDS
+)
+
+export const [gnosisSafe, gnosisSafeHooks] = initializeConnector<GnosisSafe>((actions) => new GnosisSafe(actions, true))
 
-let networkLibrary: Web3Provider | undefined
-export function getNetworkLibrary(): Web3Provider {
-  return (networkLibrary = networkLibrary ?? getLibrary(network.provider))
+export const [walletConnect, walletConnectHooks] = initializeConnector<WalletConnect>(
+  (actions) =>
+    new WalletConnect(actions, {
+      rpc: INFURA_NETWORK_URLS,
+      qrcode: true,
+    }),
+  ALL_SUPPORTED_CHAIN_IDS
+)
+
+export const [fortmatic, fortmaticHooks] = initializeConnector<EIP1193>(
+  (actions) => new EIP1193(actions, new Fortmatic(process.env.REACT_APP_FORTMATIC_KEY).getProvider())
```

Comment:
the one thing that's a little unfortunate about this approach is that it loads the fortmatic sdk for all users rather than dynamically on import like all the other connectors. not a huge deal as it's prety small: https://bundlephobia.com/package/fortmatic@2.4.0

one easy but somewhat hacky fix would be make a fake 1193 connector that dynamically imports the fortmatic sdk on the first request and then caches it, but that's optional

Reply:
why would it be dramatic? it's a tiny package. happy to do it tho

Reply:
https://bundlephobia.com/package/fortmatic@2.4.0

Reply:
i need to manually make a fortmatic connector to implement connectEagerly properly anyways, so will handle it when i do that